src/main/java/spireMapOverhaul/zones/smithsFolly/events/RansackedForge.java

```diff
@@ -0,0 +1,162 @@
+                        // Increase potion slots
+                        AbstractDungeon.player.potionSlots += 1;
+                        AbstractDungeon.player.potions.add(new PotionSlot(AbstractDungeon.player.potionSlots - 1));
+
+                        // Clear existing rewards and add one Blessing of the Forge potion
+                        AbstractDungeon.getCurrRoom().rewards.clear();
+                        AbstractDungeon.getCurrRoom().rewards.add(new RewardItem(new BlessingOfTheForge()));
+
+                        // Open the combat reward screen
+                        AbstractDungeon.combatRewardScreen.open();
+                        AbstractDungeon.getCurrRoom().rewardPopOutTimer = 0.0F;
+
+                        AbstractDungeon.combatRewardScreen.rewards.removeIf(reward -> reward.type == RewardItem.RewardType.CARD);
+
+                        // Update event text and options
+                        this.imageEventText.updateBodyText(DESCRIPTIONS[2]);
+                        this.imageEventText.updateDialogOption(0, OPTIONS[3]); // Text for closing the event
+                        this.imageEventText.clearRemainingOptions();
+                        this.screen = CurScreen.COMPLETE;
+                        break;
+
+                    case 2:
+                        // Leave
+                        this.imageEventText.updateBodyText(DESCRIPTIONS[3]);
+                        break;
+                }
+                this.imageEventText.updateDialogOption(0, OPTIONS[5]);
+                this.imageEventText.clearRemainingOptions();
+                this.screen = CurScreen.COMPLETE;
+                break;
+            case COMPLETE:
+                openMap();
+                break;
+        }
+    }
+
+    private enum CurScreen {
+        INTRO,
+        COMPLETE
+    }
+    private void handleSearingBlowOption() {
+        if (AbstractDungeon.player.masterDeck.findCardById(SearingBlow.ID) != null) {
+            // Upgrade Searing Blow four times
+            upgradeSearingBlow(4);
+            this.imageEventText.updateBodyText(DESCRIPTIONS[1]);
+        } else if (AbstractDungeon.actNum == 3 && hasStrike) {
+            // Replace all Strikes with Searing Blow+4
+            replaceStrikesWithSearingBlow();
+            this.imageEventText.updateBodyText(DESCRIPTIONS[4]);
+        } else {
+            // Obtain Searing Blow+4
+            AbstractCard searingBlow = CardLibrary.getCard(SearingBlow.ID).makeCopy();
+            searingBlow.upgrade();
+            searingBlow.upgrade();
+            searingBlow.upgrade();
+            searingBlow.upgrade();
+            AbstractDungeon.effectList.add(new ShowCardAndObtainEffect(searingBlow, (float)Settings.WIDTH / 2.0F, (float)Settings.HEIGHT / 2.0F));
+            this.imageEventText.updateBodyText(DESCRIPTIONS[5]);
+        }
+    }
+
+    private void upgradeSearingBlow(int times) {
+        for (AbstractCard card : AbstractDungeon.player.masterDeck.group) {
```

Comment:
This would upgrade multiple searing blows if you have more than one

Reply:
That is correct. I don't think this is an issue. Let's say you start Act 1 and Smith's Folly appears. You can *choose* to take every Searing Blow you see (which won't be many because it's an uncommon) for the sweet upgrades, but there's only a CHANCE of the event showing up, and if it doesn't, you're stuck hauling around a bunch of Searing Blows that are about as bad as strikes.

Reply:
Hm, fair point. I guess it could make for a fun decision